### PR TITLE
Remove duplicate mv feature state values and add unique constraint

### PR DESCRIPTION
**Pull request by matthewelwell**
This PR sets out to fix the issue of duplicate multivariate feature state values. This is currently manifesting to users in 2 ways: 

 1. The dashboard may show incorrect data as it will only show the percentage value for a single feature option, which one it shows is based on the order in which the data is returned. This may also mean that updating the values might not behave as expected. 
 2. They may get unexpected results when using MV flags as the dashboard may not show them correct data.

This PR fixes this by implementing 2 migrations:

 1. The first one removes all existing duplicate multivariate feature state values
 2. The second adds a unique constraint to prevent future dupicates being created

What this PR doesn't do is prevent the route cause of the duplicates, however, we should now get a nasty unhandled server error if one is created which sentry will pick up and we should be able to diagnose what is causing the duplicates to be created in the first place (which may in fact be related to #429).

### CODE REVIEW

File: api/features/multivariate/migrations/0002_add_unique_constraint_for_mv_feature_states.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    # This query essentially groups the MV feature state values by the feature state
+    # and MV feature option, giving us the duplicates and annotates them with the max
+    # id of that group of duplicates so we can exclude it later, obviously those that
+    # are not duplicates will therefore be excluded because they will be annotated
+    # with their own id.
+    max_id_objects = MultivariateFeatureStateValue.objects.values(
+        "feature_state", "multivariate_feature_option"
+    ).annotate(max_id=Max("id"))
+
+    max_ids = [obj.max_id for obj in max_id_objects]
```
Comment (gagantrivedi) [inline api/features/multivariate/migrations/0002_add_unique_constraint_for_mv_feature_states.py]: max_ids = [obj["max_id"] for obj in max_id_objects]

**Review by gagantrivedi — CHANGES_REQUESTED** [review]

File: api/features/multivariate/migrations/0002_add_unique_constraint_for_mv_feature_states.py (hunk 1 above)
Comment (matthewelwell) [reply api/features/multivariate/migrations/0002_add_unique_constraint_for_mv_feature_states.py]: I think both are valid?

Comment (gagantrivedi) [reply api/features/multivariate/migrations/0002_add_unique_constraint_for_mv_feature_states.py]: Nope, values returns a dict

```
In [3]: from django.db.models import Max                                                                                                                                                                                                                                                                                    
                                                                                                                                                                                                                                                                                                                            
In [4]:  max_id_objects = MultivariateFeatureStateValue.objects.values(                                                                                                                                                                                                                                                     
   ...:         "feature_state", "multivariate_feature_option"                                                                                                                                                                                                                                                              
   ...:     ).annotate(max_id=Max("id"))                                                                                                                                                                                                                                                                                    
                                                                                                                                                                                                                                                                                                                            
In [5]:                                                                                                                                                                                                                                                                                                                     
   ...:     max_ids = [obj.max_id for obj in max_id_objects]                                                                                                                                                                                                                                                                
---------------------------------------------------------------------------                                                                                                                                                                                                                                                 
AttributeError                            Traceback (most recent call last)                                                                                                                                                                                                                                                 
<ipython-input-5-b4d261973939> in <module>                                                                                                                                                                                                                                                                                  
----> 1 max_ids = [obj.max_id for obj in max_id_objects]                                                                                                                                                                                                                                                                    
                                                                                                                                                                                                                                                                                                                            
<ipython-input-5-b4d261973939> in <listcomp>(.0)                                                                                                                                                                                                                                                                            
----> 1 max_ids = [obj.max_id for obj in max_id_objects]                                                                                                                                                                                                                                                                    
                                                                                                                                                                                                                                                                                                                            
AttributeError: 'dict' object has no attribute 'max_id'                                                                                                                                                                                                                                                                     



In [6]: max_id_objects
Out[6]: <QuerySet [{'feature_state': 172, 'multivariate_feature_option': 2, 'max_id': 19}, {'feature_state': 155, 'multivariate_feature_option': 3, 'max_id': 9}, {'feature_state': 187, 'multivariate_feature_option': 6, 'max_id': 22}, {'feature_state': 171, 'multivariate_feature_option': 5, 'max_id': 15}, {'feature_state': 154, 'multivariate_feature_option': 4, 'max_id': 13}, {'feature_state': 171, 'multivariate_feature_option': 1, 'max_id': 1}, {'feature_state': 186, 'multivariate_feature_option': 8, 'max_id': 27}, {'feature_state': 188, 'multivariate_feature_option': 6, 'max_id': 23}, {'feature_state': 186, 'multivariate_fe
ature_option': 7, 'max_id': 24}, {'feature_state': 154, 'multivariate_feature_option': 3, 'max_id': 14}, {'feature_state': 153, 'multivariate_feature_option': 4, 'max_id': 10}, {'feature_state': 173, 'multivariate_feature_option': 1, 'max_id': 3}, {'feature_state': 173, 'multivariate_feature_option': 5, 'max_id': 1
7}, {'feature_state': 100, 'multivariate_feature_option': 9, 'max_id': 32}, {'feature_state': 12, 'multivariate_feature_option': 9, 'max_id': 30}, {'feature_state': 188, 'multivariate_feature_option': 8, 'max_id': 29}, {'feature_state': 153, 'multivariate_feature_option': 3, 'max_id': 7}, {'feature_state': 172, 'multivariate_feature_option': 5, 'max_id': 18}, {'feature_state': 13, 'multivariate_feature_option': 9, 'max_id': 31}, {'feature_state': 171, 'multivariate_feature_option': 2, 'max_id': 4}, '...(remaining elements truncated)...']>




```

**Comment by matthewelwell** [comment]
I've addressed your comment above @gagantrivedi but I'd like to write a migration test to confirm that what I think is happening, is indeed happening.

**Comment by matthewelwell** [comment]
@gagantrivedi this is ready for review again now that I've added the migration test.

**Review by gagantrivedi — APPROVED** [review]

**Pull request merged**